tools/clang/lib/SPIRV/SpirvEmitter.h

```diff
@@ -1248,6 +1248,26 @@ class SpirvEmitter : public ASTConsumer {
   /// Returns a function scope parameter with the same type as |param|.
   SpirvVariable *createFunctionScopeTempFromParameter(const ParmVarDecl *param);
 
+  /// Returns a vector of SpirvInstruction that is the decompostion of `inst`
+  /// into scalars. This is recursive. For example, a struct of a 4 element
+  /// vector will return 4 scalars.
+  std::vector<SpirvInstruction *> decomposeToScalars(SpirvInstruction *inst);
+
+  /// Returns a spirv instruction with the value of the given type and layout
+  /// rule that is obtained by assigning each scalar in `type` to corresponding
+  /// value in `scalars`. This is the inverse of `decomposeToScalars`.
+  SpirvInstruction *
+  GenerateFromScalars(QualType type, std::vector<SpirvInstruction *> &scalars,
```

Comment:
nit: Switch function name to lowerCamelCase (I know we're inconsistent with the style guide in general, but might as well be consistent with our inconsistency within the file)

Same for Splat below